### PR TITLE
Fixed logic with returned screamId. Refactored to use enums.

### PR DESCRIPTION
**Pull request by tylerleamon**
When UpdateScreamSound() was called, the random number generated would return a NoScream SoundID. The two resulting if statements would be executed, causing a SoundId::NoScream to be read as Null instead. This Null would cause the UpdateScreamSound() to be called continually on the same drop until a Scream was generated from RNG. The resulting code fix assumes one scream check per drop.

Also updated some case statements for enum use.

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
Presumably, this fixes #11293?

**Comment by Gymnasiast** [comment]
Could you add a changelog entry?
`Fix: [#11293]: Guests scream on every descent, no matter how small.`

**Comment by Gymnasiast** [comment]
This will also need a network version bump, I think.

**Comment by tylerleamon** [comment]
> Presumably, this fixes #11293?

Not necessarily. This fix is to prevent multiple scream checks on the same drop. That fix was a request to change the parameters that considered a valid scream during a drop.

**Comment by tupaschoal** [comment]
I think it fixes #5451 actually

**Comment by AaronVanGeffen** [comment]
Thanks, I knew there was an older bug, but couldn't find it while on my phone!

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -5511,10 +5511,12 @@ void Vehicle::UpdateSound()
                 screamId = UpdateScreamSound();
                 if (screamId == SoundId::NoScream)
                     screamId = SoundId::Null;
-                if (screamId == SoundId::Null)
+                else if (screamId == SoundId::Null)
                     goto loc_6D7A97;
                 break;
             }
+            break;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:5518]: Revert this break.

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -5511,10 +5511,12 @@ void Vehicle::UpdateSound()
                 screamId = UpdateScreamSound();
                 if (screamId == SoundId::NoScream)
                     screamId = SoundId::Null;
-                if (screamId == SoundId::Null)
+                else if (screamId == SoundId::Null)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:5514]: agreed that this looks correct

**Comment by duncanspumpkin** [comment]
Okay I checked vanilla assembly. Your if / else looks correct and fixes a mistake (looking at the assembly the jump this represents is very similar in number to 0x6d7a97 so I think when we implemented we misread it) but the extra break would deviate from vanilla. This code is quite different in assembly so its hard to follow completely but it definitely goes to `0x6d7a97` after the `if ((vehicleEntry->flags & VEHICLE_ENTRY_FLAG_RIDERS_SCREAM))`.

**Comment by duncanspumpkin** [comment]
I'll need to create an update bpb1000Ticks replay to make the ci happy. Don't worry about that I'll fix it.

**Comment by duncanspumpkin** [comment]
i've tried to be clever and do two replays at once but that of course does not work. I'll fix this tomorrow.
